### DISCUSSION TITLE
Authelia occasionally blocks connections on iOS only

### DISCUSSION THREAD

**Discussion by lilacpixel** [authelia/authelia#8069] [Troubleshooting] [answered]
Recently, Authelia has started intermittently throwing errors when attempting to access a gated resource--but *only* on iOS devices. The front end will display an "incorrect username or password" error several seconds after submitting the login form. The debug logs vary, but a few things tend to show up.

Generally, on the first login attempt, the user won't be passed in correctly:

```
time="2024-10-21T03:55:08Z" level=debug msg="Check authorization of subject username= groups= ip=[ip] and object https://service.mydomain.com/ (method GET)."
time="2024-10-21T03:55:08Z" level=info msg="Access to https://service.mydomain.com/ (method GET) is not authorized to user <anonymous>, responding with status code 302 with location redirect to https://auth.mydomain.com/?rd=https%3A%2F%2Fservice.mydomain.com%2F&rm=GET" method=GET path=/api/authz/forward-auth remote_ip=[ip]
```

Intermittent request timeouts when communicating with the reverse proxy (Caddy):

```
time="2024-10-21T03:55:24Z" level=error msg="Request timeout occurred while handling request from client." error="read tcp [authelia]:9091->[caddy]:51052: i/o timeout" method=GET path=/ remote_ip=[caddy] stack="github.com/authelia/authelia/v4/internal/server/handlers.go:86 CreateDefaultServer.handleError.func2\ngithub.com/valyala/fasthttp@v1.56.0/server.go:2884             (*Server).writeErrorResponse\ngithub.com/valyala/fasthttp@v1.56.0/server.go:2314             (*Server).serveConn\ngithub.com/valyala/fasthttp@v1.56.0/workerpool.go:225          (*workerPool).workerFunc\ngithub.com/valyala/fasthttp@v1.56.0/workerpool.go:197          (*workerPool).getCh.func1\nruntime/asm_amd64.s:1700                                       goexit" status_code=408
```

(Authelia and Caddy are both in Docker containers and share a network.)

After the first attempt, Authelia seems to authenticate the user properly, but the logs stop here, and the same error message is returned on the front end.

```
time="2024-10-21T04:01:35Z" level=debug msg="Mark 1FA authentication attempt made by user 'user'" method=POST path=/api/firstfactor remote_ip=[ip]
time="2024-10-21T04:01:35Z" level=debug msg="Successful 1FA authentication attempt made by user 'user'" method=POST path=/api/firstfactor remote_ip=[ip]
time="2024-10-21T04:01:35Z" level=debug msg="Check authorization of subject username=user groups=admins ip=[ip] and object https://service.mydomain.com/ (method GET)."
time="2024-10-21T04:01:35Z" level=debug msg="Required level for the URL https://service.mydomain.com/ is one_factor" method=POST path=/api/firstfactor remote_ip=[ip]
time="2024-10-21T04:01:35Z" level=debug msg="Redirection URL https://service.mydomain.com/ is safe" method=POST path=/api/firstfactor remote_ip=[ip]
```

The same user can authenticate and access the same service immediately on the PC. It's *just* iOS devices that do this, and not consistently. Currently, my iPad is affected, but my iPhone can still connect. A few days ago, my spouse's iPad and iPhone were both locked out in the same fashion.

This has only been happening for a week or two. I didn't change any configuration before the problem started, but it does look like Authelia got a version bump on the 6th.

Any ideas on where I might look for the source of the problem? Increasing the server timeout values didn't help, and I'm not sure what else to try.

Thanks in advance!

**Comment by james-d-elliott** [comment]
The error indicates the client is closing the connection before sending the full request. If you tell us the versions you were on previously and are on now then we can probably determine if a change on our end affected this, but nothing obvious has changed within any of the logic that handles that endpoint.

**Reply by lilacpixel** [reply]
> The error indicates the client is closing the connection before sending the full request. If you tell us the versions you were on previously and are on now then we can probably determine if a change on our end affected this, but nothing obvious has changed within any of the logic that handles that endpoint.

Thanks for responding! Looks like the previous Authelia was version 4.38.10, and the current version I'm running is 4.38.16.

**Comment by hendrik1120** [comment]
Does this issue still persist?
I checked these versions of macOS 15.0.1, iPadOS 17.7, iOS 18.0.1 and I can't replicate it.
From my experience, request timeouts can sometimes occur on Apple devices when they go into deep sleep.
My best guess would be that bad Wi-Fi could result in bad frontend ux, but that's a stretch.

**Comment by lilacpixel** [comment, marked as answer]
The issue does persist, but it looks like it's happening with the combination of iOS 18 and Caddy (more specifically, quic-go in Caddy) and not Authelia itself. Should be fixed in Caddy 2.9.0!

https://caddy.community/t/caddy-http-3-quic-and-the-new-apple-software-today/25660

I'll go ahead and close this out--thanks for the replies! 😊
